sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PreparePubsubWriteDoFn.java

```diff
@@ -193,16 +190,6 @@ public void process(
           .add("pubsub", "topic", PubsubClient.topicPathFromPath(topic).getDataCatalogSegments());
       reportedLineage = topic;
     }
-    // TODO: Remove this check once Dataflow's native sink supports ordering keys.
-    if (!allowOrderingKey && !Strings.isNullOrEmpty(message.getOrderingKey())) {
```

Comment:
could periodically log that the ordering key is present but will not be used?

We could clear the ordering key in this case so that we don't affect batching of publishing unnecessarily.

Reply:
What do you think is a good cadence here? Once per bundle may be too frequent for streaming (where I heard bundles can be very small). 

Maybe once per DoFn? It would be logged across multiple threads

Reply:
A warning log once per-dofn seems ok to surface what is going on without being too spammy.

Reply:
> We could clear the ordering key in this case so that we don't affect batching of publishing unnecessarily

Good call. I think clearing the ordering key would require us to reconstruct the whole pubsub message which may be a lil costly. Instead, I set a default empty string during batching when `withOrderingKeys()` is not set.